pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTVariableDeclaratorId.java

```diff
@@ -63,6 +63,10 @@ public boolean isExceptionBlockParameter() {
         return jjtGetParent().jjtGetParent() instanceof ASTTryStatement;
     }
 
+    public boolean isFormalParameter() {
```

Comment:
I just noticed that this method would also return true for exception variables in a catch block, since catch blocks use a FormalParameter node (related to #998), which might not be expected. I think the method should filter those out, and be documented properly.